sycl/source/detail/device_image_impl.hpp

```diff
@@ -278,21 +278,38 @@ class device_image_impl {
                                                             Descriptors.size());
         unsigned LocalOffset = 0;
         while (It != End) {
-          // Make sure that alignment is correct in blob.
-          const unsigned OffsetFromLast = /*Offset*/ It[1] - LocalOffset;
-          BlobOffset += OffsetFromLast;
+          unsigned int ID = It[0];
+          unsigned int CompositeOffset = It[1];
+          unsigned int Size = It[2];
+
+          if (MBinImage->supportsSpecConstants()) {
```

Comment:
IIRC this will not be enough to make the distinction. The check will always return true for SPIR-V binaries, but even they may be using emulated spec-consts if `-spec-const=rt` was passed to sycl-post-link.

I am not certain there is a good way to check which version of spec-constants were generated. It would be nice if there was a way to mark the data entry, but I don't believe there is currently. An alternative is to use the existing padding representation, i.e. through a spec-const entry with `uint32_t` max ID (see below).

Reply:
That's a very good point, I've had another look at this and using the padding elements I was able to make it work with no changes in the runtime at all, the current patch should reflect that.

Reply:
Great! In that case, runtime changes LGTM! 😄